app/src/main/java/io/appium/uiautomator2/model/UiObjectElement.java

```diff
@@ -128,7 +128,7 @@ public By getBy() {
     }
 
     public void clear() throws UiObjectNotFoundException {
-        element.clearTextField();
+        element.setText("");
```

Comment:
shall we check if this is really a password field first?

Reply:
`UiObject2.clear()` always perform `setText("")`. I think it can be a bit confusing if the behavior depends on the type of element and locator.

Reply:
I'm just wondering if it does not generate some additional events or something like that as a part of the method call. If you think that's safe to replace the current call and you tested it then no problems.

Reply:
There is no any difference for API>19, except [checking element text length](https://android.googlesource.com/platform/frameworks/uiautomator/+/android-support-test/src/main/java/android/support/test/uiautomator/UiObject.java#693)